### PR TITLE
feat: add next-previous component

### LINKED ISSUE #32: Next/Prev

### PR DESCRIPTION
**Pull request by vpicone**
This is a total re-write of the `NextPrevious` component. It is automatically added to the homepage and default templates and requires no intervention from developers.

Todo: 

- [x] Generate category data from yaml
- [x] Generate tab data from frontmatter
- [x] New styles
- [x] Split into a data and a presentational component
- [x] Add to homepage

closes #32

### CODE REVIEW

File: packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.module.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+.link-container {
+  min-height: rem(128px);
+  display: flex;
+  .link {
+    color: $carbon--white-0;
+    padding: 1rem;
+    text-decoration: none;
+    width: 50%;
+    transition: 0.07s background ease;
```
Comment (alisonjoseph) [inline packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.module.scss]: We should be able to use one of the fancy new duration tokens from carbon here https://www.carbondesignsystem.com/guidelines/motion/basics#duration-tokens

`$duration--fast-01` most likely 

Also, would be good to flatten this out and not nest the selectors, we had a big discussion w/ the Carbon team around this. 

```
.link-container {
...
}

.link-container .link {

}
```

Comment (vpicone) [reply packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.module.scss]: went for moderate--02 to match the IDL site

Comment (alisonjoseph) [reply packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.module.scss]: @vpicone lets double check w/ @shixiedesign on motion token for this

File: packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.module.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  margin-left: auto;
+}
+
+.direction {
+  @include carbon--type-style('body-short-01');
+}
+
+.name {
+  @include carbon--type-style('expressive-heading-03', true);
+  padding-bottom: 3rem;
```
Comment (alisonjoseph) [inline packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.module.scss]: Do we need this padding-bottom? It's making the buttons too tall at mobile breakpoints. (should also be `$spacing-09` for 3rem)

**Comment by alisonjoseph** [comment]
closes #32

**Pull request merged**